stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/SQLDatasourceUtils.java

```diff
@@ -280,18 +280,35 @@ public static String createJDBCDbOptions(String propertiesBeginSymbol, String se
         return dbOptionsStringJoiner.toString();
     }
 
+    public static BError getSQLDatabaseError(Context context, SQLException sqlException, String messagePrefix) {
+        String detailedErrorMessage =
+                sqlException.getMessage() != null ? sqlException.getMessage() : Constants.DATABASE_ERROR_MESSAGE;
+        int vendorCode = sqlException.getErrorCode();
+        String sqlState = sqlException.getSQLState();
+        return getSQLConnectorError(context, messagePrefix + detailedErrorMessage, vendorCode, sqlState);
+    }
 
-    public static BError getSQLConnectorError(Context context, Throwable throwable, String messagePrefix) {
+    public static BError getSQLApplicationError(Context context, Throwable throwable, String messagePrefix) {
         String detailedErrorMessage =
-                throwable.getMessage() != null ? throwable.getMessage() : Constants.DATABASE_ERROR_MESSAGE;
+                throwable.getMessage() != null ? throwable.getMessage() : Constants.APPLICATION_ERROR_MESSAGE;
         return getSQLConnectorError(context, messagePrefix + detailedErrorMessage);
     }
 
-    public static BError getSQLConnectorError(Context context, String detailedErrorMessage) {
-        BMap<String, BValue> sqlClientErrorDetailRecord = BLangConnectorSPIUtil
-                .createBStruct(context, Constants.SQL_PACKAGE_PATH, Constants.DATABASE_ERROR_DATA_RECORD_NAME,
-                        detailedErrorMessage);
-        return BLangVMErrors.createError(context, true, BTypes.typeError, Constants.DATABASE_ERROR_CODE,
+    public static BError getSQLConnectorError(Context context, String detailedErrorMessage, Object... values) {
```

Comment:
Remove this common method and move the respective logic to the `getSQLDatabaseError` and `getSQLApplicationError` methods directly. Having Object... and if checks within the method adds no value

Reply:
This getSQLConnectorError method was used in this line [1]. I haven't remove this method in this https://github.com/ballerina-platform/ballerina-lang/pull/15841/commits/b3309499bc1a5fd8905e6b505b094ad552e25f1d . But i have moved database error logic to getSQLDatabaseError method

[1] https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/actions/Close.java#L56